File: src/main/java/net/igneo/icv/entity/voidSpike/VoidSpikeModel.java
package net.igneo.icv.entity.voidSpike;

import net.igneo.icv.ICV;
import net.minecraft.resources.ResourceLocation;
import software.bernie.geckolib.model.GeoModel;

public class VoidSpikeModel extends GeoModel<VoidSpikeEntity> {
    private final ResourceLocation model = new ResourceLocation(ICV.MOD_ID, "geo/placeholder.geo.json");
    private final ResourceLocation texture = new ResourceLocation(ICV.MOD_ID, "textures/entity/placeholder.png");
    private final ResourceLocation animations = new ResourceLocation(ICV.MOD_ID, "animations/placeholder.animation.json");
    @Override
    public ResourceLocation getModelResource(VoidSpikeEntity surfWaveEntity) {
        return model;
    }

    @Override
    public ResourceLocation getTextureResource(VoidSpikeEntity surfWaveEntity) {
        return texture;
    }

    @Override
    public ResourceLocation getAnimationResource(VoidSpikeEntity surfWaveEntity) {
        return animations;
    }
}

File: src/main/java/net/igneo/icv/entity/leggings/voidSpike/VoidSpikeModel.kt
package net.igneo.icv.entity.leggings.voidSpike

import net.igneo.icv.ICV
import net.minecraft.resources.ResourceLocation
import software.bernie.geckolib.model.GeoModel

class VoidSpikeModel : GeoModel<VoidSpikeEntity?>() {
    private val model = ResourceLocation(ICV.MOD_ID, "geo/placeholder.geo.json")
    private val texture = ResourceLocation(ICV.MOD_ID, "textures/entity/placeholder.png")
    private val animations = ResourceLocation(ICV.MOD_ID, "animations/placeholder.animation.json")

    override fun getModelResource(surfWaveEntity: VoidSpikeEntity?): ResourceLocation {
        return model
    }

    override fun getTextureResource(surfWaveEntity: VoidSpikeEntity?): ResourceLocation {
        return texture
    }

    override fun getAnimationResource(surfWaveEntity: VoidSpikeEntity?): ResourceLocation {
        return animations
    }
}
